src/python/pants/backend/python/tasks/pytest_run.py

```diff
@@ -563,7 +563,12 @@ def _get_failed_targets_from_junitxml(self, junitxml, targets, pytest_rootdir):
             pytest_relpath = testcase.getAttribute('file')
             relsrc = os.path.join(buildroot_relpath, pytest_relpath)
```

Comment:
In the failing case, was the computed `relsrc` bad? ie, was it not a valid file? Or was the file correct, but `relsrc_to_target` didn't contain the right file or target?

Reply:
The `file` attribute in `testcase` element inside the XML file(`junitxml`) does not have a valid filename/file path  (`file="&lt;string&gt;"`)
`testcase.getAttribute('file')` returns `<string>`.
Hence `relsrc` is `.pants.d/<string>`